### ISSUE TITLE
Search: The link text in the filters should be more descriptive

### ISSUE THREAD

**Issue by XiuJuanXia** [mmiron-HC/WST-RSAMS#120]
URL: https://hcdev.openplus.ca/en/search/site
User Agent: Chrome
Language: Both

Issue Description: The link text in the filters should be more descriptive. For example, the first one is "Recall 2074 link". What is the 2074 for?
![image](https://user-images.githubusercontent.com/88776404/137160061-998a857b-9663-4f52-bf3d-ad067a0d284f.png)

Steps to reproduce: Visual check

Expected results: The link text should be more descriptive.

**Comment by ndolter** [comment]
Validated in both languages that the screen reader now reads 'Recall, 2077 results available link'
